tap_salesforce/__init__.py

```diff
@@ -105,28 +104,26 @@ def main_impl():
 def sync(
     sf: Salesforce,
     stream: Stream,
-    stream_id: str,
-    fields: Dict[str, Field],
-    replication_key: str,
+    table: Table,
+    fields: List[str],
     start_time: datetime,
     end_time: datetime,
     limit: Optional[int] = None,
 ):
     try:
-        for raw_record in sf.get_records(
-            stream_id,
+        for record in sf.get_records(
+            table,
             fields,
-            replication_key,
             start_time,
             end_date=end_time,
             limit=limit,
         ):
-            record = transform_record(raw_record, fields)
-
-            stream.write_record(record, stream_id)
-            state_value = record[replication_key]
 
-            stream.set_stream_state(stream_id, replication_key, state_value)
+            stream.write_record(record, table.name)
+            state_value = datetime.fromisoformat(
+                record[table.replication_key][: -len("+0000")]
+            )
```

Comment:
Totally dig converting to `datetime` here, but I'd suggest using `datetime.strptime` with the expected format, rather than this. Would make it easier to read.

```suggestion
            state_value = datetime.strptime(record[table.replication_key], "%Y-%m-%dT%H:%M:%S")
```

Alternatively, I see that [`datetime.fromisoformat` should handle the offset by itself.](https://docs.python.org/3/library/datetime.html#datetime.datetime.fromisoformat)

Reply:
it couldn't, because the datetime looks like this: `2023-01-25T02:30:05.000+0000`